src/legacy/config/config-set.spec.ts

```diff
@@ -424,7 +425,7 @@ describe('tsCacheDir', () => {
       std: '9.10.2',
       opt: '2.0.2',
     }
-    const mock: jest.MockInstance<string | undefined, [string]> = jest
+    const mock: MockInstance<(s: string) => string | undefined> = jest
       .mocked(getPackageVersion)
       .mockImplementation((moduleName: string) => realVersions[moduleName])
```

Comment:
Feels like `jest.mocked()` should return correct type. Did you try simply:

```suggestion
    const mock = jest.mocked(getPackageVersion).mockImplementation((moduleName: string) => realVersions[moduleName])
```